eclass/java-pkg-simple.eclass

```diff
@@ -543,6 +543,22 @@ java-pkg-simple_src_compile() {
 
 		local tmp_source=${JAVA_PKG_WANT_SOURCE} tmp_target=${JAVA_PKG_WANT_TARGET}
 
+		# Initialize a variable to track the highest key
+		highest_key=-1
+
+		# Loop through the keys of the associative array
+		for key in "${!JAVA_RELEASE_SRC_DIRS[@]}"; do
+		    # Compare the numeric value of the key
+		    if [[ key > highest_key ]]; then
+		        highest_key="$key"
+		    fi
+		done
+
+		local jdk="$(depend-java-query --get-lowest "${DEPEND}")"
+		if [[ "${jdk#1.}" -lt "${highest_key}" ]]; then
+			die || "Wrong DEPEND, needs higher version in >=virtual/jdk"
```

Comment:
nice! how about putting `${highest_key}` in the error message? also there's the redundant `||` for the `die` statements. it should be just `die <message>`.